validation-test/stdlib/Collection/LazyFilterCollection.swift

```diff
@@ -0,0 +1,78 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyFilterCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyFilterCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+// Test collection as sequence using value types as elements.
+CollectionTests.addSequenceTests(
+  makeSequence: { (elements: [OpaqueValue<Int>]) in
+    elements.lazy.filter { _ in return true }
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeSequenceOfEquatable: { (elements: [MinimalEquatableValue]) in
+    elements.lazy.filter { _ in return true }
+  },
+  wrapValueIntoEquatable: identityEq,
+  extractValueFromEquatable: identityEq
+)
+
+// Test collection as sequence using reference types as elements.
+CollectionTests.addSequenceTests(
+  makeSequence: { (elements: [LifetimeTracked]) in
+    elements.lazy.filter { _ in return true }
```

Comment:
You should wrap elements into a minimal collection of appropriate kind (forward and bidirectional).